concourse/scripts/kipper.sh

```diff
@@ -0,0 +1,358 @@
+    local target_ec2_instances_count
+    target_ec2_instances_count=$(get_length "${target_ec2_instances_ids}")
+    if [[ ${previous_ec2_instances_count} -ne ${target_ec2_instances_count} ]] || [[ "${previous_ec2_instances_count}" = '' ]]; then
+      echo "ec2 instances ${target_ec2_state}: ${target_ec2_instances_count}/${initial_ec2_instances_count}"
+      previous_ec2_instances_count=${target_ec2_instances_count}
+      sleep 3
+    fi
+  done
+}
+
+# action db instances state change, loop waiting for target state
+manage_db_instances_state() {
+  local initial_db_instances
+  initial_db_instances=$(get_db_instances "${initial_db_state}")
+  local initial_db_instances_ids
+  initial_db_instances_ids=$(get_db_instances_ids "${initial_db_instances}")
+  local initial_db_instances_count
+  initial_db_instances_count=$(get_length "${initial_db_instances_ids}")
+  if [[ "${initial_db_instances_count}" = '0' || "${initial_db_instances_count}" = '' ]]; then
+    echo "${env_arg} db instances are already ${past_tense}"
+    return 0
+  fi
+  echo "found ${initial_db_instances_count} db instances to ${action}..."
+  trigger_db_instances_state_update "${initial_db_instances_ids}" "${action}"
+  echo "issued db instances state change: ${initial_db_state}->${target_db_state}..."
+  local target_db_instances_count='0'
+  local previous_db_instances_count=''
+  while [[ ${target_db_instances_count} -lt ${initial_db_instances_count} ]]; do
+    local target_db_instances
+    target_db_instances=$(get_db_instances "${target_db_state}" "${initial_db_instances_ids}")
+    local target_db_instances_ids
+    target_db_instances_ids=$(get_db_instances_ids "${target_db_instances}")
+    local target_db_instances_count
+    target_db_instances_count=$(get_length "${target_db_instances_ids}")
+    if [[ ${previous_db_instances_count} -ne ${target_db_instances_count} ]] || [[ "${previous_db_instances_count}" = '' ]]; then
+      echo "db instances ${target_db_state}: ${target_db_instances_count}/${initial_db_instances_count}"
+      previous_db_instances_count=${target_db_instances_count}
+      sleep 3
+    fi
+  done
+}
+
+print_error() {
+  local string
+  string="${1:-}"
+  local level
+  level="${2:-FATAL}"
+  echo -e "${level}: ${string}" >&2
+  if [[ "${level}" = 'FATAL' ]]; then
+    exit 1
+  fi
+}
+
+check_dependencies() {
+  echo 'checking dependencies...'
+  local bash_ver
+  bash_ver=$(echo "${BASH_VERSION}" | cut -d'.' -f1)
+  if [[ "${bash_ver}" -lt 4 ]]; then
+    print_error 'requires bash version >= 4'
+  fi
+  if ! command -v bosh >/dev/null 2>&1; then
+    print_error 'missing bosh cli'
+  fi
+  if ! command -v jq >/dev/null 2>&1; then
+    print_error 'missing jq'
+  fi
+  if ! command -v aws >/dev/null 2>&1; then
+    print_error 'missing aws cli'
+  fi
+}
+
+validate_initial_state() {
+  echo 'validating initial state...'
+  if ! bosh vms >/dev/null 2>&1; then
+    print_error 'not logged in to bosh, or cf not deployed'
```

Comment:
Suggest it only gives up if it can't contact either the cf deployment or one named `prometheus` or one named `app-autoscaler` before it concludes it can't do anything useful with bosh.

(even then it might be useful for it to try turning off RDS instances if that's easy)

Reply:
implemented - im testing now

Reply:
Awesome. Was thinking about this last night and I reckon the only bosh deployment you can really rely on existing is `concourse`. Well, it may not exist, but at that point we're beyond all hope because we wouldn't be able to revive it.

Reply:
ok changes applied and tested